src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertyNamesShouldNotMatchGetMethods.cs

```diff
@@ -50,61 +49,55 @@ private static void AnalyzeSymbol(SymbolAnalysisContext context)
+                // Want to look for properties named the same as the method sans 'Get'
                 identifier = symbol.Name.Substring(3);
             }
             else
             {
-                //if method name doesn't start with Get exit
+                // Exit if the method name doesn't start with 'Get'
                 return;
             }
 
-            //boolean variable used to exit out of the inner and outer for loops
-            var matchFound = false;
-
-            //get the collection of declaring and base types
-            var types = symbol.ContainingType.GetBaseTypesAndThis();
-
-            //iterate through the collection to find match
-            foreach (INamedTypeSymbol type in types)
+            // Iterate through all declared types, including base
+            foreach (INamedTypeSymbol type in symbol.ContainingType.GetBaseTypesAndThis())
             {
-                ImmutableArray<ISymbol> membersFound = type.GetMembers(identifier);
-                if (membersFound != null && membersFound.Length > 0)
+                Diagnostic diagnostic = null;
+
+                foreach (ISymbol member in type.GetMembers(identifier))
                 {
-                    //found a match
-                    foreach (ISymbol member in membersFound)
+                    // If the declared type is a property, was a matching method found?
+                    if (symbol.Kind == SymbolKind.Property && member.Kind == SymbolKind.Method)
                     {
-                        //valid matches are...
-                        //when property from declaring type matches with method present in declaring type - this is covered by the LHS of OR condition below
-                        //when property from declaring type matches with method present in one of the base types - this is covered by the LHS of OR condition below
-                        //when method from declaring type matches with property present in one of the base types - this is covered by the RHS of OR condition below
-                        if ((symbol.Kind == SymbolKind.Property && member.Kind == SymbolKind.Method) ||
-                            (symbol.Kind == SymbolKind.Method && member.Kind == SymbolKind.Property && symbol.ContainingType != type))
-                        {
-                            //match found and break out of inner for loop
-                            matchFound = true;
-                            break;
-                        }
+                        diagnostic = Diagnostic.Create(Rule, symbol.Locations[0], symbol.Name, identifier);
+                        break;
                     }
 
-                    //if no match found iterate to next in outer for loop
-                    if (!matchFound)
+                    // If the declared type is a method, was a matching property found? Although this
+                    // check seems redundant, it's the only way to catch violations of this rule when the
```

Comment:
Ahh, I was thinking if I checked all properties against all methods, there was no need to do the reverse. But, I overlooked the fact that I wouldn't be checking all properties. If Child was in its own file and defined a method called `GetTitle`, I wouldn't iterate through the properties of Parent to find `Title`. Thus, I'll remove my comment.

To answer your question, where the warnings would be reported depends on how those classes are distributed into files. If they were each in their own file, then Child.cs would report `Name` and Grandchild.cs would report `GetAge`.